From f48b4f6766fcaa193652fcbe6ea0bb29f92e45aa Mon Sep 17 00:00:00 2001
From: Pan Xiuli <xiuli.pan@intel.com>
Date: Wed, 23 Sep 2015 10:35:20 +0800
Subject: Driver: fix the annoying "Failed to release userptr..." error message

It is a drm related bug. As the drm driver changed the time to free their test
userptr to bufmgr destroy(30921483c70c6939f017476eac13da6aa26b3b3c), we need
anothr order to release our driver to make sure the test userptr can be freed
with a valid fd.

Signed-off-by: Pan Xiuli <xiuli.pan@intel.com>
Reviewed-by: "Yang, Rong R" <rong.r.yang@intel.com>

diff --git a/src/intel/intel_driver.c b/src/intel/intel_driver.c
index 507c910..035a103 100644
--- a/src/intel/intel_driver.c
+++ b/src/intel/intel_driver.c
@@ -85,8 +85,6 @@ intel_driver_delete(intel_driver_t *driver)
   if (driver == NULL)
     return;
 
-  if (driver->bufmgr)
-    drm_intel_bufmgr_destroy(driver->bufmgr);
   cl_free(driver);
 }
 
@@ -257,6 +255,10 @@ intel_driver_open(intel_driver_t *intel, cl_context_prop props)
 static void
 intel_driver_close(intel_driver_t *intel)
 {
+  //Due to the drm change about the test usrptr, we need to destroy the bufmgr
+  //befor the driver was closed, otherwise the test usrptr will not be freed.
+  if (intel->bufmgr)
+    drm_intel_bufmgr_destroy(intel->bufmgr);
 #ifdef HAS_X11
   if(intel->dri_ctx) dri_state_release(intel->dri_ctx);
   if(intel->x11_display) XCloseDisplay(intel->x11_display);
-- 
cgit v0.10.2

